### PR TITLE
feat: enable XMPP WebSocket on mobile

### PR DESCRIPTION
**Pull request by paweldomas**
Enables XMPP WebSocket on mobile.

It would be good to land https://github.com/jitsi/lib-jitsi-meet/pull/1303 before merging this PR.

### CODE REVIEW

**Review by damencho — DISMISSED** [review]

**Comment by saghul** [comment]
Sorry I dropped the ball on this one. I'm somewhat worried things may go sideways and I want to find a way to switch back without an app update. We might need a new (temporary) setting.

**Comment by suleymangunduzz** [comment]
Hello there, do you have a final desicion about that development ? We want to implement that on our mobile clients but when we test the app somehow in some cases ( which we are not 100% sure about ) app disconnects from room.

**Comment by madf12** [comment]
Tried to use websocket and found that CONFERENCE_FAILED event not fires on android when internet connection is off, but ios is ok.

**Comment by saghul** [comment]
Time to push this one forward :-)

**Comment by saghul** [comment]
I have amended the original commit with support for the keep-alive URL. I'll continue to test this.

**Review by saghul — APPROVED** [review]

File: react/features/base/connection/actions.native.js [hunk 1]
```diff
@@ -275,10 +275,17 @@ function _constructOptions(state) {
     // redux store.
     const options = _.cloneDeep(state['features/base/config']);
 
-    // Normalize the BOSH URL.
-    let { bosh } = options;
+    let { bosh, websocket } = options;
```
Comment (damencho) [inline react/features/base/connection/actions.native.js:278]: in connection.js we are getting rid of .bosh config, is there a reason to have it here?

Comment (damencho) [reply react/features/base/connection/actions.native.js:278]: Aaa we remove the passing it to ljm, but we still have it in config.js, is this right?

Comment (saghul) [reply react/features/base/connection/actions.native.js:278]: Correct!

**Review by damencho — APPROVED** [review]

**Review by saghul — APPROVED** [review]

**Pull request merged**